src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
+
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int RESIZE_INDEX = 2;
+    private Node [] buckets = new Node[DEFAULT_CAPACITY];
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        if (size > buckets.length * LOAD_FACTOR) {
+            resize();
+        }
+        int index = getHashCode(key) % buckets.length;
+        if (buckets[index] == null) {
+            buckets[index] = new Node(getHashCode(key), key, value, null);
```

Comment:
getHashCode(key) - you use this several times, let's create a separate variable and store it there, so you could use it everywhere you need without executing this method every time

Reply:
Yap, it makes sense. Done